core/src/main/java/org/wso2/testgrid/core/TestPlanExecutor.java

```diff
@@ -339,6 +339,177 @@ private void runScenarioTests(TestPlan testPlan, DeploymentCreationResult deploy
+            String filePath = configChangeSetRepoPath.getPath();
+            String fileName = filePath.substring(filePath.lastIndexOf('/') + 1);
+            String[] initShellCommand = {
+                    "mkdir repos",
+                    "cd repos && curl -LJO " +
+                            testPlan.getConfigChangeSetRepository() + "/archive/master.tar.gz   &>/dev/null",
+                    "cd repos && tar xvzf " + fileName + "-master.tar.gz  &>/dev/null",
+                    "chmod -R 755 repos/" + fileName + "-master/config-sets/ &>/dev/null"
+            };
+            return applyShellCommandOnAgent(testPlan, initShellCommand);
+        } catch (MalformedURLException e) {
+            logger.warn("Error parsing scenario path " + e);
+            return false;
+        }
+    }
+
+    /**
+     * Revert back changes did in initConfigChangeSet
+     *
+     * @param testPlan  The test plan
+     * @return          True if execution success. Else, false
+     */
+    private boolean deInitConfigChangeSet(TestPlan testPlan) {
+        String[] deInitShellCommand = {
+                "rm -rf repos"
+        };
+        return applyShellCommandOnAgent(testPlan, deInitShellCommand);
+    }
+
+    /**
+     * Send an array of shell command to an agent
+     *
+     * @param testPlan          The test plan
+     * @param initShellCommand  Array of shell command
+     * @return  True if execution success. Else, false
+     */
+    private boolean applyShellCommandOnAgent(TestPlan testPlan, String[] initShellCommand) {
+        String tinkererHost = ConfigurationContext.getProperty(
+                ConfigurationContext.ConfigurationProperties.DEPLOYMENT_TINKERER_REST_BASE_PATH);
+        // Get authentication detail from config.properties and encode with BASE64
+        String authenticationString = ConfigurationContext.getProperty(
+                ConfigurationContext.ConfigurationProperties.DEPLOYMENT_TINKERER_USERNAME) + ":" +
+                ConfigurationContext.getProperty(
+                        ConfigurationContext.ConfigurationProperties.DEPLOYMENT_TINKERER_PASSWORD);
+        String authenticationToken = "Basic " + Base64.getEncoder().encodeToString(
+                authenticationString.getBytes(StandardCharsets.UTF_8));
+        try {
+            Content agentResponse = Request.Get(tinkererHost + "/agents")
+                    .addHeader(HttpHeaders.AUTHORIZATION, authenticationToken).
+                            execute().returnContent();
+            try {
+                // Todo use with Agent class with jackson mapper
+                JSONArray agentArray = new JSONArray(agentResponse.asString());
+                for (int index = 0; index < agentArray.length(); index++) {
+                    JSONObject agent = agentArray.getJSONObject(index);
+                    if (agent.getString("testPlanId").equals(testPlan.getId())) {
```

Comment:
Is it guaranteed to return an not null value even if agent.getString("testPlanId") does not exists?